src/Microsoft.Tye.Hosting/ProcessRunner.cs

```diff
@@ -330,6 +330,39 @@ private static string GetExePath(string projectFilePath)
             return null;
         }
 
+        public Task HandleStaleReplica(ReplicaEvent replicaEvent)
+        {
+            var processStatus = (ProcessStatus)replicaEvent.Replica;
+            var pid = processStatus.Pid ?? -1;
```

Comment:
nitpick  - there's no real advantage to using `-1` as a sentinel value instead of null. `-1` is what languages use when they don't have a nullable type 😆